src/extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,76 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Kiota.Abstractions.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="IEnumerable"/>
+    /// </summary>
+    public static class IEnumerableExtensions
+    {
+        /// <summary>
+        /// Converts an <see cref="IEnumerable{T}"/> to a <see cref="List{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the collection.</typeparam>
+        /// <param name="e">The enumerable to convert.</param>
+        /// <returns>A <see cref="List{T}"/> containing the elements of the enumerable.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the input enumerable is null.</exception>
+        public static List<T> AsList<T>(this IEnumerable<T> e)
+        {
+            if (e is null) throw new ArgumentNullException(nameof(e), "Input collection cannot be null.");
```

Comment:
what do you think about returning null instead to preserve the behaviour of what we have today with null propagation?